turbinia/workers/docker.py

```diff
@@ -0,0 +1,97 @@
+# -*- coding: utf-8 -*-
+# Copyright 2015 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Task for running docker-explorer."""
+
+from __future__ import unicode_literals
+
+import json
+import logging
+import os
+import subprocess
+
+from turbinia import TurbiniaException
+from turbinia.evidence import DockerContainer
+from turbinia.processors import mount_local
+from turbinia.workers import TurbiniaTask
+
+log = logging.getLogger('turbinia')
+
+
+class DockerContainersEnumerationTask(TurbiniaTask):
+  """Enumerates Docker containers on Linux"""
+
+  def GetContainers(self, evidence):
+    """Lists the containers from an input Evidence.
+
+    Args:
+      evidence: the input Evidence.
+
+    Returns:
+      a list(dict) containing information about the containers found.
+
+    Raises:
+      TurbiniaException: when the docker-explorer tool failed to run.
+    """
+
+    mount_path = evidence.local_path
+    if type(evidence).__name__ == 'RawDisk':
+      mount_path = mount_local.PreprocessMountDisk(
+          evidence.loopdevice_path, evidence.mount_partition)
+
+    docker_dir = os.path.join(mount_path, 'var', 'lib', 'docker')
+    containers_info = None
+    docker_explorer_command = [
+        'sudo', '/usr/local/bin/de.py', '-r', docker_dir, 'list',
```

Comment:
this is now in gift PPA where canonical path is /usr/bin.

There is also a packaged module, and we could make turbinia depend on it (instead of raising if we can't find it) WDYT?

Reply:
Yeah if there is a packaged module, let's make Turbinia depend on that.  Otherwise if it's a system dependency it is related to this issue:
https://github.com/google/turbinia/issues/312

Reply:
SGTM.
I'm in the process of cleaning up the code of DE to make using the module a bit easier, will then rework this part.

Reply:
This is done